### DISCUSSION TITLE
Using pnpm as node manager while still using other package managers such as yarn

### DISCUSSION THREAD

**Discussion by sjiep** [pnpm/pnpm#4995] [Q&A]
I'm in the process of switching to pnpm to manage my node versions. However, I still have projects that should be maintained by yarn as a package manager. 

In order to use yarn, I installed it as a global package, i.e. `pnpm add -g yarn` under node version 16. This gives me the `yarn` executable and I can use it. However when I node switch node version with `pnpm env use --global 12`, the `yarn` executable is still called from the node@16, as you can see from the file contents of `.../.local/share/pnpm/yarn`.

```
.../.local/share/pnpm/nodejs/16.16.0/bin/node"  "$basedir/global/5/node_modules/yarn/bin/yarn.js" "$@"
``` 

This causes some problems between projects that are on different node versions.

The quick fix I found for now, is to call yarn with `pnpm dlx yarn`, which actually works and resolves the issue, but it does feel a bit hacky.

Is there a better way to use yarn as package manager while have pnpm as node manager? One possible solution I could think of is that globally installed packages (`pnpm add -g`) are bound to current active node environment.

**Comment by zkochan** [comment]
Why is it a problem that Yarn is executed by Node.js 16?

I guess we could add something like `pnpm add -g yarn --use-active-node`. But I am not sure it would be useful in any other case. That is why I wonder why Yarn has issues.

**Comment by sjiep** [comment]
Perhaps very specific, but I have a package that runs a post-script on `npm install` which fails when run node@16 and not on node@12. 

I think I resolve the problem by updating the corresponding packages that cause the problem. However I would also assume that when running yarn (installed through `pnpm add -g yarn`) it would use node version it was installed for. 

Alternatively as you suggest on the "active node", but perpaps the added package also has a dependency on the node version it was installed for so that might also bring issues for some packages.

I'm guessing it would be a very low priority feature.
